scripts/benchmark.py

```diff
@@ -0,0 +1,320 @@
+
+CLUSTER_CONFIGS = [
+    {
+        "name": "small-local-cluster-replica",
+        "per_worker_cores": 2,
+        "workers": 4,
+    },
+    {
+        "name": "large-local-cluster-replica",
+        "per_worker_cores": 2,
+        "workers": 16,
+    },
+    {
+        "name": "small-worker-distributed-cluster",
+        "per_worker_cores": 1,
+        "workers": 32,
+    },
+    {
+        "name": "many-small-worker-distributed-cluster",
+        "per_worker_cores": 1,
+        "workers": 128,
+    },
+    {
+        "name": "many-standard-worker-distributed-cluster",
+        "per_worker_cores": 4,
+        "workers": 32,
+    },
+]
+
+###############################################################################
+
+# Args
+
+
+class Args(argparse.Namespace):
+    def __init__(self):
+        self.__parse()
+
+    def __parse(self):
+        # Setup parser
+        p = argparse.ArgumentParser(
+            prog="benchmark_aicsimageio",
+            description=(
+                "Run read time benchmarks for aicsimageio against other common image "
+                "readers. The benchmark dataset can be downloaded using the "
+                "download_test_resources script with the specific hash: "
+                "06489d5501d1003e41046a3cb8a57887f04e75b30e23416ff60da8b87b80beb6"
+            )
+        )
+
+        # Arguments
+        p.add_argument(
+            "--save_path",
+            default="benchmark/results.json",
+            type=Path,
+            help="Path to save the generated benchmark JSON file.",
+        )
+        p.add_argument(
+            "--upload",
+            action="store_true",
+            help="Should the results be uploaded to Quilt."
+        )
+        p.add_argument(
+            "--debug",
+            action="store_true",
+            help="Show traceback if the script were to fail.",
+        )
+
+        # Parse
+        p.parse_args(namespace=self)
+
+
+###############################################################################
+
+def _run_benchmark(
+    resources_dir: Path,
+    extensions: List[str],
+    non_aicsimageio_reader: List[Callable],
+    iterations: int = 3,
+):
+    # Collect files matching the extensions provided
+    files = []
+    for ext in extensions:
+        files += list(resources_dir.glob(ext))
+
+    # Run reads for each file and store details in results
+    per_file_results = []
+    for file in files:
+        info_read = aicsimageio.AICSImage(file)
+        yx_planes = np.prod(info_read.size("STCZ"))
+        for reader in [aicsimageio.imread, non_aicsimageio_reader]:
+            reader_path = f"{reader.__module__}.{reader.__name__}"
+            read_durations = []
+            for i in tqdm(range(iterations), desc=f"{reader_path}: {file.name}"):
+                start = datetime.utcnow()
+                reader(str(file))
+                read_durations.append((datetime.utcnow() - start).total_seconds())
```

Comment:
I might recommend [`time.perf_counter`](https://docs.python.org/3/library/time.html#time.perf_counter) here instead of `datetime` to ensure you have the highest available clock resolution.

Reply:
Totally agree...  if you even need sub-second timing for this stuff?

Reply:
Alright if I update it in the #104 PR?